gatsby/src/pages/search.tsx

```diff
@@ -1,14 +1,278 @@
+	return text
+		? page > 1
+			? `Page ${page} | ${text} | `
+			: `${text} | `
+		: page > 1
+		? `Page ${page} | `
+		: "";
+}
+
+const SearchPage: React.FC<SearchPageProps> = ({
+	location,
+}: SearchPageProps) => {
+	const [data, setData] = useState<SearchResults | null>(null);
+	const [error, setError] = useState<Error>();
+
+	useEffect(() => {
+		setData(null);
+		fetch("/api/search" + window.location.search)
+			.then(data => data.json())
+			.then(results => setData(results as SearchResults))
+			.catch(e => setError(e));
+	}, [location.search]);
+
 	return (
 		<Layout>
-			<SEO title="Search" />
 			<h1>Search</h1>
+			{!error && !data && <div>Loading</div>}
+			{data && data.failed && <div>Failed to load results</div>}
+			{data && !data.failed && <Results {...data} />}
+			{error?.message && (
+				<Alert type="error">
+					<pre>
+						{error.name}
+						<br />
+						{error.message}
+						<br />
+						{error.stack}
+					</pre>
+				</Alert>
+			)}
 		</Layout>
 	);
 };
 
+const Results: React.FC<SearchResults> = ({
+	firstResult,
+	resultCount,
+	finalSearchText,
+	originalSearch,
+	documents,
+	pageSize,
+	pagerLinks: { next, previous },
+}) => {
+	const currentPage = Math.ceil(firstResult / pageSize);
+	const totalPages = Math.ceil(resultCount / pageSize);
+	const paginationProps = {
+		nextPageLink: next?.fullUrl
+			? { destination: "/" + next?.fullUrl, elementType: Link }
+			: undefined,
+		previousPageLink: previous?.fullUrl
+			? { destination: "/" + previous?.fullUrl, elementType: Link }
+			: undefined,
+		currentPage,
+		totalPages,
+	};
+
+	return (
+		<>
+			<SEO
+				title={`${titleString(finalSearchText, currentPage)}Search results`}
+				noIndex={true}
+			/>
+			<ResultSummary
+				resultCount={resultCount}
+				finalSearchText={finalSearchText}
+				originalSearch={originalSearch?.searchText}
+			/>
+			{documents.length > 0 && <ResultsList documents={documents} />}
+			{resultCount > pageSize && <SimplePagination {...paginationProps} />}
+		</>
+	);
+};
+
+interface ResultsSummary {
+	resultCount: number;
+	finalSearchText: string;
+	originalSearch: string;
+}
+
+const ResultSummary: React.FC<ResultsSummary> = ({
+	resultCount,
+	finalSearchText,
+	originalSearch,
+}) => {
+	if (resultCount === 0) {
+		return <NoResults searchText={finalSearchText} />;
+	}
+
+	return (
+		<>
+			<p>
+				{originalSearch && (
+					<>
+						Your search for <b>{originalSearch}</b> returned no results
+						<br />
+					</>
+				)}
+				{resultCount} {resultCount === 1 ? "result" : "results"}
+				{finalSearchText && (
+					<>
+						{" "}
+						for <b>{finalSearchText}</b>
+					</>
+				)}
+			</p>
+		</>
+	);
+};
+
+interface ResultsList {
+	documents: Document[];
+}
+
+const ResultsList: React.FC<ResultsList> = ({ documents }) => (
+	<ul className="list--unstyled">
+		{documents?.map(({ id, title, pathAndQuery, teaser }) => (
+			<li key={id}>
+				<Card
+					summary={parser(teaser)}
```

Comment:
The parser is handy because it turns it into a string where we've required a string in the card types - I think you mentioned restricting the summary to a string so people couldn't pass in more complex html? The only thing we're really looking to allow here is encoded characters. I was an advocate for allowing people to pass in any HTML they wanted into the summary so could change the card to allow that with documentation to keep to a string / text content to abide to the guidelines?

Reply:
Yeah maybe that's more flexible - people still need to check it looks right and is accessible, we can't do everything for them!

Reply:
Cool I'll raise a card for that against NDS